---
title: "2024_03_04_VCF_class"
format: html
editor: source
---

This workbook will go through the "Genomic Variants with Bioconductor course from `https://rockefelleruniversity.github.io/RU_GenomicVariants/index.html`

# Install packages

```{r}
# install.packages('BiocManager')
# BiocManager::install('RockefellerUniversity/RU_GenomicVariants',subdir='genomicvariants')
# 
# install.packages('BiocManager')
# BiocManager::install('methods')
# BiocManager::install('ggplot2')
# BiocManager::install('rmarkdown')
# BiocManager::install('dplyr')
# BiocManager::install('knitr')
# BiocManager::install('RCurl')
# BiocManager::install('VariantAnnotation')
# BiocManager::install('BSgenome.Hsapiens.UCSC.hg19')
# BiocManager::install('TxDb.Hsapiens.UCSC.hg19.knownGene')
# BiocManager::install('SNPlocs.Hsapiens.dbSNP144.GRCh37')
# BiocManager::install('GenomicFeatures')
# BiocManager::install('maftools')
# BiocManager::install('NMF')
# BiocManager::install('BiocGenerics')
# BiocManager::install('R.utils')
# BiocManager::install('pheatmap')
# BiocManager::install('stringr')
# BiocManager::install('testthat')
# BiocManager::install('yaml')
```

# Load Libraries

```{r}
suppressPackageStartupMessages({
library(methods)
library(tidyverse)
library(rmarkdown)
library(RCurl)
library(VariantAnnotation)
library(BSgenome.Hsapiens.UCSC.hg38)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(SNPlocs.Hsapiens.dbSNP144.GRCh37)
library(GenomicFeatures)
library(maftools)
library(NMF)
library(BiocGenerics)
library(R.utils)
library(pheatmap)
library(stringr)
library(testthat)
library(yaml)
})
```

# Define path variables

```{r}
data.dir <- "data/raw_data/"
```

# Test run with a single sample

```{r}
s76.c <- readVcf(file = paste0(data.dir, "76_C_S47.haplotypecaller.filtered_snpEff.ann.vcf"), "hg38")

s76.c
```

```{r}
VariantAnnotation::header(s76.c)
header(temp)
# geno(header(temp))

header
```












