<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2024_05_03_VCF_Code_Review</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="2024_05_03_VCF_Code_Review_files/libs/clipboard/clipboard.min.js"></script>
<script src="2024_05_03_VCF_Code_Review_files/libs/quarto-html/quarto.js"></script>
<script src="2024_05_03_VCF_Code_Review_files/libs/quarto-html/popper.min.js"></script>
<script src="2024_05_03_VCF_Code_Review_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2024_05_03_VCF_Code_Review_files/libs/quarto-html/anchor.min.js"></script>
<link href="2024_05_03_VCF_Code_Review_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2024_05_03_VCF_Code_Review_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2024_05_03_VCF_Code_Review_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2024_05_03_VCF_Code_Review_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2024_05_03_VCF_Code_Review_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full">

<main class="content column-page" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">2024_05_03_VCF_Code_Review</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>This notebook will detail the process of annotating VCF files using ANNOVAR run in a powershell conda environment, then using RStudio to read in, interrogate, and visualize those results with maftools.</p>
<p>Background on the samples: There are 6 whole genome sequencing Variant Call Format (VCF) files, 3 from bone marrow biopsies of Down Syndrome patients with B-cell acute lymphoblastic leukemia (BALL), and 3 samples of immortalized cell lines created from those biopsy samples.</p>
<p>Essentially, we obtained small portions of the biopsy samples and put them into continuous cell culture and the cancerous cells continued to grow and expand.</p>
<p>Currently there are no established cell lines of Down Syndrome BALL to use as models in experiments to help discover treatments.</p>
<p>We performed the RNAseq on the initial biopsies (labeled “Pt” in the file names) and the resulting cell lines (labeled “CL”) that proliferated in culture to compare them to each other.</p>
<p>The post-doc doing this work is writing a paper that details and characterizes the cell lines, and an important part of doing that is comparing their “signatures”, both genetic and phenotypic, to the initial biopsy samples. We’re comparing signatures, showing similarities and noting any contrasts in an effort to establish the new cell lines as a model to use for research of BALL in Down Syndrome patients.</p>
<section id="libraries" class="level1">
<h1>Libraries</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(maftools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vcfR)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.3.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">14000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="directories" class="level1">
<h1>Directories</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"data/raw_data/"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>results.dir <span class="ot">&lt;-</span> <span class="st">"2024_05_03_results/"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>plots.dir <span class="ot">&lt;-</span> <span class="st">"2024_05_03_results/plots/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="vcf-files---brief-overview" class="level1">
<h1>VCF files - Brief Overview</h1>
<p>VCF is a common output format for variant call programs due to its simplicity and scalability.</p>
<p>Unfortunately for researchers, it is not very informative in its native form.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>vcf.df <span class="ot">&lt;-</span> vcfR<span class="sc">::</span><span class="fu">read.vcfR</span>(<span class="at">file =</span> <span class="fu">paste0</span>(data.dir, <span class="st">"DS-1-Pt.haplotypecaller.filtered_snpEff.ann.vcf"</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>vcf.df <span class="ot">&lt;-</span> <span class="fu">vcfR2tidy</span>(vcf.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Look at the data</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(vcf.df<span class="sc">$</span>fix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 78%">
<col style="width: 0%">
<col style="width: 0%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">ChromKey</th>
<th style="text-align: left;">CHROM</th>
<th style="text-align: right;">POS</th>
<th style="text-align: left;">ID</th>
<th style="text-align: left;">REF</th>
<th style="text-align: left;">ALT</th>
<th style="text-align: right;">QUAL</th>
<th style="text-align: left;">FILTER</th>
<th style="text-align: left;">AC</th>
<th style="text-align: left;">AF</th>
<th style="text-align: right;">AN</th>
<th style="text-align: right;">BaseQRankSum</th>
<th style="text-align: right;">DP</th>
<th style="text-align: right;">ExcessHet</th>
<th style="text-align: right;">FS</th>
<th style="text-align: right;">InbreedingCoeff</th>
<th style="text-align: left;">MLEAC</th>
<th style="text-align: left;">MLEAF</th>
<th style="text-align: right;">MQ</th>
<th style="text-align: right;">MQRankSum</th>
<th style="text-align: right;">QD</th>
<th style="text-align: right;">ReadPosRankSum</th>
<th style="text-align: right;">SOR</th>
<th style="text-align: left;">ANN</th>
<th style="text-align: left;">LOF</th>
<th style="text-align: left;">NMD</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">14020</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
<td style="text-align: right;">37.32</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">1.00</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">18.66</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0.693</td>
<td style="text-align: left;">T|downstream_gene_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene||n.<em>384G&gt;A|||||384|,T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.</em>3349G&gt;A|||||3349|,T|non_coding_transcript_exon_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript|3/3|n.1268C&gt;T||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">14464</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">T</td>
<td style="text-align: right;">194.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">-1.625</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">7.49</td>
<td style="text-align: right;">1.407</td>
<td style="text-align: right;">0.269</td>
<td style="text-align: left;">T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>55A&gt;T|||||55|,T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.</em>2905T&gt;A|||||2905|,T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|11/11|n.1291T&gt;A||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">15817</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">T</td>
<td style="text-align: right;">334.64</td>
<td style="text-align: left;">FS</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.031</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">30.797</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6.20</td>
<td style="text-align: right;">1.250</td>
<td style="text-align: right;">5.363</td>
<td style="text-align: left;">T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>1408G&gt;T|||||1408|,T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.</em>1552C&gt;A|||||1552|,T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|9/11|n.1198C&gt;A||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">15820</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">T</td>
<td style="text-align: right;">725.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1.694</td>
<td style="text-align: right;">56</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">25.072</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13.44</td>
<td style="text-align: right;">2.184</td>
<td style="text-align: right;">4.328</td>
<td style="text-align: left;">T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>1411G&gt;T|||||1411|,T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.</em>1549C&gt;A|||||1549|,T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|9/11|n.1195C&gt;A||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">15903</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">GC</td>
<td style="text-align: right;">128.10</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">1.00</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">1.00</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">32.02</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">3.258</td>
<td style="text-align: left;">GC|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>1494_</em>1495insC|||||1495|,GC|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.*1465dupG|||||1465|,GC|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|9/11|n.1111dupG||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">15922</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">113.75</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">-0.366</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">0.500</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">16.25</td>
<td style="text-align: right;">0.792</td>
<td style="text-align: right;">1.179</td>
<td style="text-align: left;">G|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>1513A&gt;G|||||1513|,G|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.</em>1447T&gt;C|||||1447|,G|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|9/11|n.1093T&gt;C||||||</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>It’s hard to know what a change in chromosome 1 at position 14,020 from C to T means in terms of effect on the amino acid coding or what gene it relates to. That’s where ANNOVAR and maftools come in. They will allow us to add more informative annotation to these variants and create some plots that help us better understand the landscape of variants in our samples.</p>
</section>
<section id="annovar" class="level1">
<h1>ANNOVAR</h1>
<section id="pc-setup---powershell-conda-environment" class="level2">
<h2 class="anchored" data-anchor-id="pc-setup---powershell-conda-environment">PC setup - Powershell conda environment</h2>
<p>I work on a PC, so I don’t have a native environment to run ANNOVAR, which is written in Perl. I need to create a conda environment with Perl installed to run ANNOVAR.</p>
<p>The following code chunks were based off of the tutorial recommended by the ANNOVAR website - <a href="https://github.com/WGLab/Workshop_Annotation" class="uri">https://github.com/WGLab/Workshop_Annotation</a></p>
<p>I started by downloading the <a href="https://www.anaconda.com/download">Anaconda Distribution Software</a> and installing it.</p>
<p>After opening the Navigator, I select “Powershell Prompt.”</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/anaconda.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>Here’s a look at the powershell terminal.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/terminal.png" class="img-fluid figure-img" width="667"></p>
</figure>
</div>
<p>This starts a terminal in my <code>Users</code> folder, which is on a different hard drive than the rest of my R data. For convenience I created a few new directories to store relevant files and scripts here.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>mkdir conda_projects</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>cd .\conda_projects\</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>mkdir 2023_10_17_Dallas_SNP_vcf_files</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>cd .\2023_10_17_Dallas_SNP_vcf_files\</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>mkdir data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>I copied my VCF files into the <code>data</code> folder.</p>
<p>Now I can create my conda environment. I will name it “annotation.”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>conda create <span class="sc">--</span>name annotation</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>conda activate annotation</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/conda%20activate.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>The <em>(annotation)</em> at the front of the terminal shows that we are in the newly created conda environment. Now to install Perl.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>conda install perl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Select “y” when prompted. Once it is complete you can verify the install.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check version of perl installed</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>perl <span class="sc">-</span>v</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/perl%20version.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>Now we can download and install ANNOVAR. You need to enter your name and e-mail on the <a href="https://www.openbioinformatics.org/annovar/annovar_download_form.php">ANNOVAR website</a> to get the zip file. Unzip the file and put the contents into your project folder, which is <em>2023_10_17_Dallas_SNP_vcf_files/</em> in this walkthrough. The contents should look something like this:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/annovar%20scripts.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>After that, we need to download and install extra libraries. Libraries available are listed <a href="https://annovar.openbioinformatics.org/en/latest/user-guide/download/">here</a>. Each library has a different version (hg19, hg38, etc), so check the genome version used to generate your VCF files.</p>
<p>Libraries needed:</p>
<ul>
<li><p>refGene</p></li>
<li><p>cytoBand</p></li>
<li><p>exac03</p></li>
<li><p>avsnp147</p></li>
<li><p>dbnsfp30a</p></li>
</ul>
<p>The <code>annotate_variation.pl</code> script has the download function. The command syntax is:</p>
<p><code>perl annotate_variation.pl -buildver (genome version) -downdb -webfrom annovar (library name) (path)</code></p>
<p>The parts in parentheses are to be filled in by the user. I am also creating a folder called <code>humandb</code> to store the libraries and reduce clutter in my project folder.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>perl annotate_variation.pl <span class="sc">-</span>buildver hg38 <span class="sc">-</span>downdb <span class="sc">-</span>webfrom annovar refGene humandb<span class="sc">/</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>perl annotate_variation.pl <span class="sc">-</span>buildver hg38 <span class="sc">-</span>downdb <span class="sc">-</span>webfrom annovar cytoBand humandb<span class="sc">/</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>perl annotate_variation.pl <span class="sc">-</span>buildver hg38 <span class="sc">-</span>downdb <span class="sc">-</span>webfrom annovar exac03 humandb<span class="sc">/</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>perl annotate_variation.pl <span class="sc">-</span>buildver hg38 <span class="sc">-</span>downdb <span class="sc">-</span>webfrom annovar avsnp147 humandb<span class="sc">/</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>perl annotate_variation.pl <span class="sc">-</span>buildver hg38 <span class="sc">-</span>downdb <span class="sc">-</span>webfrom annovar dbnsfp30a humandb<span class="sc">/</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Unzip the libraries and extract the files. Most of the libraries automatically start their file names with the genome version, except for the Cytoband.txt file. In this case, I renamed it “hg38_cytoBand.txt”.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/annova%20libraries.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="table_annovar.pl" class="level2">
<h2 class="anchored" data-anchor-id="table_annovar.pl">table_annovar.pl</h2>
<p>Now we can run <code>table_annovar.pl</code>. I’ll put the command first, then follow it up with a table describing all the various parts.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In powershell terminal</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>perl table_annovar.pl data<span class="sc">/</span>DS<span class="dv">-1</span><span class="sc">-</span>CL.haplotypecaller.filtered_snpEff.ann.vcf humandb<span class="sc">/</span> <span class="sc">-</span>buildver hg38 <span class="sc">-</span>out hg38_results<span class="sc">/</span>DS<span class="dv">-1</span><span class="sc">-</span>CL <span class="sc">-</span>remove <span class="sc">-</span>protocol refGene,cytoBand,exac03,avsnp147,dbnsfp30a <span class="sc">-</span>operation g,r,f,f,f <span class="sc">-</span>nastring . <span class="sc">-</span>vcfinput <span class="sc">-</span>polish</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat for all 6 samples: DS-1-CL, DS-1-Pt, DS-76-CL, DS-76-Pt, DS-81-CL, DS-81-Pt</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 25%">
<col style="width: 49%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Code</th>
<th style="text-align: center;">Description</th>
<th style="text-align: center;">Other Info</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">table_annovar.pl</td>
<td style="text-align: center;">Script we are calling</td>
<td style="text-align: center;">Perl scripts end in .pl</td>
</tr>
<tr class="even">
<td style="text-align: center;">data/DS-1-CL.haplotypecaller.filtered_snpEff.ann.vcf</td>
<td style="text-align: center;">Directory and name of the input file</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">humandb/</td>
<td style="text-align: center;">Location of library files</td>
<td style="text-align: center;">refGene, CytoBand, etc.</td>
</tr>
<tr class="even">
<td style="text-align: center;">-buildver hg38</td>
<td style="text-align: center;">Flag for genome version (hg38 here)</td>
<td style="text-align: center;">Could be hg19, mm9, etc.</td>
</tr>
<tr class="odd">
<td style="text-align: center;">-out hg38_results/DS-1-CL</td>
<td style="text-align: center;">Flag for output directory (hg38_results) and file name. It will make the directory if it doesn’t exist</td>
<td style="text-align: center;">“DS-1-CL” will be the prefix on all output files</td>
</tr>
<tr class="even">
<td style="text-align: center;">-remove</td>
<td style="text-align: center;">Removes temporary files</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">-protocol refGene,cytoBand,exac03,avsnp147,dbnsfp30a</td>
<td style="text-align: center;">Protocol flag followed by all the functions we are using. Comma separated</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">-operation g,r,f,f,f</td>
<td style="text-align: center;">Operation flag specifying type of operation. Comma separated. Five protocols require five flags</td>
<td style="text-align: center;"><p>g = gene</p>
<p>r = region</p>
<p>f = filter</p></td>
</tr>
<tr class="odd">
<td style="text-align: center;">-nastring .</td>
<td style="text-align: center;">Value to display when a score is not available</td>
<td style="text-align: center;">“.” in our code</td>
</tr>
<tr class="even">
<td style="text-align: center;">-vcfinput</td>
<td style="text-align: center;">This flag specifies a VCF file as input</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">-polish</td>
<td style="text-align: center;">“Polishes” the protein notation for indels (ex: p.G12Vfs*2)</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
<p>These and more options can be found by running <code>perl table_annovar.pl –help</code> in the terminal.</p>
<p>The results from the function look like this:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/annovar%20results.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>These text results can be imported into R using maftools.</p>
</section>
</section>
<section id="maftools" class="level1">
<h1>maftools</h1>
<section id="read-in-annovar-results" class="level2">
<h2 class="anchored" data-anchor-id="read-in-annovar-results">Read in ANNOVAR results</h2>
<p>I copy/pasted the result files back into my R workspace in <code>2024_05_03_results</code>.</p>
<p>Now to read in the ANNOVAR results, which are in multianno.txt files. We can use the <code>annovarToMaf()</code> function to do this. The function also lets us input a vector of files. This way it reads and combines them in one step.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an input vector of file names</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>file.names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"2024_05_03_results/DS-1-CL.hg38_multianno.txt"</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                <span class="st">"2024_05_03_results/DS-1-Pt.hg38_multianno.txt"</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                <span class="st">"2024_05_03_results/DS-76-CL.hg38_multianno.txt"</span>,</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                <span class="st">"2024_05_03_results/DS-76-Pt.hg38_multianno.txt"</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>                <span class="st">"2024_05_03_results/DS-81-CL.hg38_multianno.txt"</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>                <span class="st">"2024_05_03_results/DS-81-Pt.hg38_multianno.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now to run <code>annovarToMaf()</code>. It has the following syntax:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/annovar%20to%20maf.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>The following code chunk provides specifies our file name vector, genome version, that they’re tab separated text files, and that they are MAF objects. <code>basename</code> provides a file name to save the resulting MAF file.</p>
<div class="cell" data-hash="2024_05_03_VCF_Code_Review_cache/html/unnamed-chunk-12_b0583921ec849354ebf7fbf6527ac08f">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ds.maf <span class="ot">&lt;-</span> <span class="fu">annovarToMaf</span>(<span class="at">annovar =</span> file.names,</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">refBuild =</span> <span class="st">"hg38"</span>,</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">MAFobj =</span> T,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">basename =</span> <span class="fu">paste0</span>(results.dir, <span class="st">"ds"</span>))</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in pre-saved MAF object</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.maf &lt;- read.maf(paste0(results.dir, "ds.maf"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/maftoannovar.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>NOTE: I am taking advantage of a built in feature of the function. MAF files typically have a column called <code>Tumor_Sample_Barcode</code>, which is the package’s version of sample ID. If that column isn’t found or specified (using <code>tsbCol = )</code>, the function uses the file name to generate an ID. It is smart enough to clip off “.hg38_multianno.txt” as well, so if you have good file names they can be used in the MAF object.</p>
<p>Take a look at the object. It is a S4 object with a few components:</p>
<ul>
<li><span class="citation" data-cites="data">@data</span> - contains the newly annotated variant data</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 9%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 32%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: left;">Chromosome</th>
<th style="text-align: right;">Start_Position</th>
<th style="text-align: right;">End_Position</th>
<th style="text-align: left;">Reference_Allele</th>
<th style="text-align: left;">Tumor_Seq_Allele2</th>
<th style="text-align: left;">Hugo_Symbol</th>
<th style="text-align: left;">Variant_Classification</th>
<th style="text-align: left;">tx</th>
<th style="text-align: left;">exon</th>
<th style="text-align: left;">txChange</th>
<th style="text-align: left;">aaChange</th>
<th style="text-align: left;">Variant_Type</th>
<th style="text-align: left;">Func.refGene</th>
<th style="text-align: left;">Gene.refGene</th>
<th style="text-align: left;">GeneDetail.refGene</th>
<th style="text-align: left;">ExonicFunc.refGene</th>
<th style="text-align: left;">AAChange.refGene</th>
<th style="text-align: left;">cytoBand</th>
<th style="text-align: left;">ExAC_ALL</th>
<th style="text-align: left;">ExAC_AFR</th>
<th style="text-align: left;">ExAC_AMR</th>
<th style="text-align: left;">ExAC_EAS</th>
<th style="text-align: left;">ExAC_FIN</th>
<th style="text-align: left;">ExAC_NFE</th>
<th style="text-align: left;">ExAC_OTH</th>
<th style="text-align: left;">ExAC_SAS</th>
<th style="text-align: left;">avsnp147</th>
<th style="text-align: left;">SIFT_score</th>
<th style="text-align: left;">SIFT_pred</th>
<th style="text-align: left;">Polyphen2_HDIV_score</th>
<th style="text-align: left;">Polyphen2_HDIV_pred</th>
<th style="text-align: left;">Polyphen2_HVAR_score</th>
<th style="text-align: left;">Polyphen2_HVAR_pred</th>
<th style="text-align: left;">LRT_score</th>
<th style="text-align: left;">LRT_pred</th>
<th style="text-align: left;">MutationTaster_score</th>
<th style="text-align: left;">MutationTaster_pred</th>
<th style="text-align: left;">MutationAssessor_score</th>
<th style="text-align: left;">MutationAssessor_pred</th>
<th style="text-align: left;">FATHMM_score</th>
<th style="text-align: left;">FATHMM_pred</th>
<th style="text-align: left;">PROVEAN_score</th>
<th style="text-align: left;">PROVEAN_pred</th>
<th style="text-align: left;">VEST3_score</th>
<th style="text-align: left;">CADD_raw</th>
<th style="text-align: left;">CADD_phred</th>
<th style="text-align: left;">DANN_score</th>
<th style="text-align: left;">fathmm-MKL_coding_score</th>
<th style="text-align: left;">fathmm-MKL_coding_pred</th>
<th style="text-align: left;">MetaSVM_score</th>
<th style="text-align: left;">MetaSVM_pred</th>
<th style="text-align: left;">MetaLR_score</th>
<th style="text-align: left;">MetaLR_pred</th>
<th style="text-align: left;">integrated_fitCons_score</th>
<th style="text-align: left;">integrated_confidence_value</th>
<th style="text-align: left;">GERP++_RS</th>
<th style="text-align: left;">phyloP7way_vertebrate</th>
<th style="text-align: left;">phyloP20way_mammalian</th>
<th style="text-align: left;">phastCons7way_vertebrate</th>
<th style="text-align: left;">phastCons20way_mammalian</th>
<th style="text-align: left;">SiPhy_29way_logOdds</th>
<th style="text-align: left;">Otherinfo1</th>
<th style="text-align: left;">Otherinfo2</th>
<th style="text-align: left;">Otherinfo3</th>
<th style="text-align: left;">Otherinfo4</th>
<th style="text-align: left;">Otherinfo5</th>
<th style="text-align: left;">Otherinfo6</th>
<th style="text-align: left;">Otherinfo7</th>
<th style="text-align: left;">Otherinfo8</th>
<th style="text-align: left;">Otherinfo9</th>
<th style="text-align: left;">Otherinfo10</th>
<th style="text-align: left;">Otherinfo11</th>
<th style="text-align: left;">Otherinfo12</th>
<th style="text-align: left;">Otherinfo13</th>
<th style="text-align: left;">ref_alt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">946228</td>
<td style="text-align: right;">946228</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">NOC2L</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_015658</td>
<td style="text-align: left;">exon16</td>
<td style="text-align: left;">c.A1862G</td>
<td style="text-align: left;">p.H621R</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">NOC2L</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">NOC2L:NM_015658:exon16:c.A1862G:p.H621R</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.116</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.584</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">0.345</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.985</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.1</td>
<td style="text-align: left;">L</td>
<td style="text-align: left;">1.03</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">-1.37</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">0.064</td>
<td style="text-align: left;">1.560</td>
<td style="text-align: left;">13.64</td>
<td style="text-align: left;">0.982</td>
<td style="text-align: left;">0.911</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">-0.988</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.106</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.707</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">5.17</td>
<td style="text-align: left;">0.991</td>
<td style="text-align: left;">0.047</td>
<td style="text-align: left;">0.974</td>
<td style="text-align: left;">0.967</td>
<td style="text-align: left;">14.183</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">2155.64</td>
<td style="text-align: left;">246</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">946228</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">2155.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.056;DP=255;ExcessHet=0.0000;FS=2.153;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=8.76;ReadPosRankSum=-1.055;SOR=0.783;ANN=C|missense_variant|MODERATE|NOC2L|ENSG00000188976|transcript|ENST00000327044.7|protein_coding|16/19|c.1862A&gt;G|p.His621Arg|1878/2757|1862/2250|621/749||,C|downstream_gene_variant|MODIFIER|SAMD11|ENSG00000187634|transcript|ENST00000618323.5|protein_coding||c.*2075T&gt;C|||||1654|</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:138,108:246:99:2163,0,3140</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">953279</td>
<td style="text-align: right;">953279</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">NOC2L</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_015658</td>
<td style="text-align: left;">exon9</td>
<td style="text-align: left;">c.A898G</td>
<td style="text-align: left;">p.I300V</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">NOC2L</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">NOC2L:NM_015658:exon9:c.A898G:p.I300V</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">0.9355</td>
<td style="text-align: left;">0.9096</td>
<td style="text-align: left;">0.9557</td>
<td style="text-align: left;">0.9126</td>
<td style="text-align: left;">0.9600</td>
<td style="text-align: left;">0.9407</td>
<td style="text-align: left;">0.9509</td>
<td style="text-align: left;">0.9183</td>
<td style="text-align: left;">rs3748597</td>
<td style="text-align: left;">1.0</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.070</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">-2.24</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">-0.39</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.76</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">0.091</td>
<td style="text-align: left;">-1.330</td>
<td style="text-align: left;">0.004</td>
<td style="text-align: left;">0.237</td>
<td style="text-align: left;">0.042</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">-0.957</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.672</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">4.28</td>
<td style="text-align: left;">0.002</td>
<td style="text-align: left;">0.016</td>
<td style="text-align: left;">0.751</td>
<td style="text-align: left;">0.143</td>
<td style="text-align: left;">7.695</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">2877.06</td>
<td style="text-align: left;">107</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">953279</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">2877.06</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=2;AF=1.00;AN=2;DP=109;ExcessHet=0.0000;FS=0.000;MLEAC=2;MLEAF=1.00;MQ=60.00;QD=26.89;SOR=0.750;ANN=C|missense_variant|MODERATE|NOC2L|ENSG00000188976|transcript|ENST00000327044.7|protein_coding|9/19|c.898A&gt;G|p.Ile300Val|914/2757|898/2250|300/749||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">1/1:0,107:107:99:2891,321,0</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">976209</td>
<td style="text-align: right;">976209</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">PERM1</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_001369898</td>
<td style="text-align: left;">exon3</td>
<td style="text-align: left;">c.A2270G</td>
<td style="text-align: left;">p.Q757R</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">PERM1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">PERM1:NM_001369898:exon3:c.A2270G:p.Q757R,PERM1:NM_001291366:exon4:c.A2336G:p.Q779R,PERM1:NM_001369897:exon4:c.A2336G:p.Q779R,PERM1:NM_001291367:exon5:c.A1994G:p.Q665R</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.611</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.001</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.004</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.316</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">-0.8</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">0.055</td>
<td style="text-align: left;">-1.383</td>
<td style="text-align: left;">0.003</td>
<td style="text-align: left;">0.409</td>
<td style="text-align: left;">0.020</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">-1.046</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.038</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.616</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">-5.39</td>
<td style="text-align: left;">0.051</td>
<td style="text-align: left;">-0.266</td>
<td style="text-align: left;">0.007</td>
<td style="text-align: left;">0.027</td>
<td style="text-align: left;">2.320</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">78.32</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">976209</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">78.32</td>
<td style="text-align: left;">SnpCluster</td>
<td style="text-align: left;">AC=2;AF=1.00;AN=2;DP=2;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;QD=33.27;SOR=0.693;ANN=C|missense_variant|MODERATE|PERM1|ENSG00000187642|transcript|ENST00000433179.3|protein_coding|3/3|c.2336A&gt;G|p.Gln779Arg|2336/3340|2336/2373|779/790||,C|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|transcript|ENST00000379409.6|protein_coding||c.*1634T&gt;C|||||1201|</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">1/1:0,2:2:6:90,6,0</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">976215</td>
<td style="text-align: right;">976215</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">PERM1</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_001369898</td>
<td style="text-align: left;">exon3</td>
<td style="text-align: left;">c.T2264C</td>
<td style="text-align: left;">p.V755A</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">PERM1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">PERM1:NM_001369898:exon3:c.T2264C:p.V755A,PERM1:NM_001291366:exon4:c.T2330C:p.V777A,PERM1:NM_001369897:exon4:c.T2330C:p.V777A,PERM1:NM_001291367:exon5:c.T1988C:p.V663A</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">0.7712</td>
<td style="text-align: left;">0.4986</td>
<td style="text-align: left;">0.7639</td>
<td style="text-align: left;">0.915</td>
<td style="text-align: left;">0.8636</td>
<td style="text-align: left;">0.7787</td>
<td style="text-align: left;">0.7042</td>
<td style="text-align: left;">0.7877</td>
<td style="text-align: left;">rs7417106</td>
<td style="text-align: left;">0.024</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.518</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">0.103</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.174</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">-2.52</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.123</td>
<td style="text-align: left;">2.802</td>
<td style="text-align: left;">21.4</td>
<td style="text-align: left;">0.951</td>
<td style="text-align: left;">0.055</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">-1.052</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.616</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">2.15</td>
<td style="text-align: left;">0.123</td>
<td style="text-align: left;">1.138</td>
<td style="text-align: left;">0.172</td>
<td style="text-align: left;">0.814</td>
<td style="text-align: left;">6.784</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">78.32</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">976215</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">78.32</td>
<td style="text-align: left;">SnpCluster</td>
<td style="text-align: left;">AC=2;AF=1.00;AN=2;DP=2;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;QD=34.96;SOR=0.693;ANN=G|missense_variant|MODERATE|PERM1|ENSG00000187642|transcript|ENST00000433179.3|protein_coding|3/3|c.2330T&gt;C|p.Val777Ala|2330/3340|2330/2373|777/790||,G|downstream_gene_variant|MODIFIER|PLEKHN1|ENSG00000187583|transcript|ENST00000379409.6|protein_coding||c.*1640A&gt;G|||||1207|</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">1/1:0,2:2:6:90,6,0</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">1053827</td>
<td style="text-align: right;">1053827</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">AGRN</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_198576</td>
<td style="text-align: left;">exon34</td>
<td style="text-align: left;">c.G5726C</td>
<td style="text-align: left;">p.S1909T</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">AGRN</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">AGRN:NM_198576:exon34:c.G5726C:p.S1909T</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">0.0119</td>
<td style="text-align: left;">0.0014</td>
<td style="text-align: left;">0.1155</td>
<td style="text-align: left;">0.0111</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.0006</td>
<td style="text-align: left;">0.0100</td>
<td style="text-align: left;">0.0005</td>
<td style="text-align: left;">rs74685771</td>
<td style="text-align: left;">0.578</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.068</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.387</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.983</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.41</td>
<td style="text-align: left;">L</td>
<td style="text-align: left;">-1.19</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">-1.4</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">0.3</td>
<td style="text-align: left;">1.732</td>
<td style="text-align: left;">14.59</td>
<td style="text-align: left;">0.502</td>
<td style="text-align: left;">0.980</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">-0.893</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.123</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.707</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">-0.779</td>
<td style="text-align: left;">-0.020</td>
<td style="text-align: left;">-1.015</td>
<td style="text-align: left;">0.985</td>
<td style="text-align: left;">0.029</td>
<td style="text-align: left;">12.707</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">587.64</td>
<td style="text-align: left;">88</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">1053827</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">587.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=0.607;DP=90;ExcessHet=0.0000;FS=9.533;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=6.68;ReadPosRankSum=-1.212;SOR=1.765;ANN=C|missense_variant|MODERATE|AGRN|ENSG00000188157|transcript|ENST00000379370.7|protein_coding|34/36|c.5726G&gt;C|p.Ser1909Thr|5779/7326|5726/6138|1909/2045||,C|upstream_gene_variant|MODIFIER|ENSG00000242590|ENSG00000242590|transcript|ENST00000418300.1|pseudogene||n.-1206G&gt;C|||||1206|</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:58,30:88:99:595,0,1343</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">1203873</td>
<td style="text-align: right;">1203873</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">TNFRSF18</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_004195</td>
<td style="text-align: left;">exon5</td>
<td style="text-align: left;">c.A697G</td>
<td style="text-align: left;">p.K233E</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">TNFRSF18</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">TNFRSF18:NM_004195:exon5:c.A697G:p.K233E,TNFRSF18:NM_148902:exon5:c.A676G:p.K226E</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.006</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.713</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">0.162</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.24</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">-1.36</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">-1.044</td>
<td style="text-align: left;">0.013</td>
<td style="text-align: left;">0.982</td>
<td style="text-align: left;">0.049</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">-0.946</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.124</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.615</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">2.07</td>
<td style="text-align: left;">0.991</td>
<td style="text-align: left;">1.051</td>
<td style="text-align: left;">0.002</td>
<td style="text-align: left;">0.003</td>
<td style="text-align: left;">2.529</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">23.64</td>
<td style="text-align: left;">6</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">1203873</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">23.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.834;DP=7;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=3.94;ReadPosRankSum=0.000;SOR=0.693;ANN=C|synonymous_variant|LOW|TNFRSF18|ENSG00000186891|transcript|ENST00000328596.10|protein_coding|4/4|c.486A&gt;G|p.Arg162Arg|486/851|486/768|162/255||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:4,2:6:31:31,0,104</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="variants.per.sample">@variants.per.sample</span> - Variant count per sample</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>variants.per.sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: right;">Variants</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: right;">12059</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: right;">11437</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-81-CL</td>
<td style="text-align: right;">11163</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-76-CL</td>
<td style="text-align: right;">10248</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-76-Pt</td>
<td style="text-align: right;">5291</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-81-Pt</td>
<td style="text-align: right;">4702</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="variant.type.summary">@variant.type.summary</span> - Breakdown of variant type per sample (Ins, Del, SNP)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>variant.type.summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: right;">DEL</th>
<th style="text-align: right;">INS</th>
<th style="text-align: right;">SNP</th>
<th style="text-align: right;">total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: right;">1071</td>
<td style="text-align: right;">518</td>
<td style="text-align: right;">10470</td>
<td style="text-align: right;">12059</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: right;">830</td>
<td style="text-align: right;">633</td>
<td style="text-align: right;">9974</td>
<td style="text-align: right;">11437</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-81-CL</td>
<td style="text-align: right;">993</td>
<td style="text-align: right;">680</td>
<td style="text-align: right;">9490</td>
<td style="text-align: right;">11163</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-76-CL</td>
<td style="text-align: right;">793</td>
<td style="text-align: right;">609</td>
<td style="text-align: right;">8846</td>
<td style="text-align: right;">10248</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-76-Pt</td>
<td style="text-align: right;">110</td>
<td style="text-align: right;">505</td>
<td style="text-align: right;">4676</td>
<td style="text-align: right;">5291</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-81-Pt</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">335</td>
<td style="text-align: right;">4273</td>
<td style="text-align: right;">4702</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="variant.classification.summary">@variant.classification.summary</span> - Breakdown of mutation type per sample (missense, frame shift, etc)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>variant.classification.summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 13%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 10%">
<col style="width: 14%">
<col style="width: 3%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: right;">Frame_Shift_Del</th>
<th style="text-align: right;">Frame_Shift_Ins</th>
<th style="text-align: right;">In_Frame_Del</th>
<th style="text-align: right;">In_Frame_Ins</th>
<th style="text-align: right;">Missense_Mutation</th>
<th style="text-align: right;">Nonsense_Mutation</th>
<th style="text-align: right;">Nonstop_Mutation</th>
<th style="text-align: right;">Translation_Start_Site</th>
<th style="text-align: right;">total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: right;">844</td>
<td style="text-align: right;">418</td>
<td style="text-align: right;">168</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">10237</td>
<td style="text-align: right;">259</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">12059</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: right;">652</td>
<td style="text-align: right;">540</td>
<td style="text-align: right;">125</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">9747</td>
<td style="text-align: right;">234</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">11437</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-81-CL</td>
<td style="text-align: right;">790</td>
<td style="text-align: right;">580</td>
<td style="text-align: right;">161</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">9280</td>
<td style="text-align: right;">232</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">11163</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-76-CL</td>
<td style="text-align: right;">619</td>
<td style="text-align: right;">519</td>
<td style="text-align: right;">133</td>
<td style="text-align: right;">65</td>
<td style="text-align: right;">8655</td>
<td style="text-align: right;">208</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">10248</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-76-Pt</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">423</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">57</td>
<td style="text-align: right;">4602</td>
<td style="text-align: right;">59</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">5291</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-81-Pt</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">263</td>
<td style="text-align: right;">66</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">4213</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">4702</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="gene.summary">@gene.summary</span> - Mutations types for each gene</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>gene.summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 6%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 3%">
<col style="width: 8%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Hugo_Symbol</th>
<th style="text-align: right;">Frame_Shift_Del</th>
<th style="text-align: right;">Frame_Shift_Ins</th>
<th style="text-align: right;">In_Frame_Del</th>
<th style="text-align: right;">In_Frame_Ins</th>
<th style="text-align: right;">Missense_Mutation</th>
<th style="text-align: right;">Nonsense_Mutation</th>
<th style="text-align: right;">Nonstop_Mutation</th>
<th style="text-align: right;">Translation_Start_Site</th>
<th style="text-align: right;">total</th>
<th style="text-align: right;">MutatedSamples</th>
<th style="text-align: right;">AlteredSamples</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">OBSCN</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">135</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">151</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">TTN</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">111</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">116</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PABPC3</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">95</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">DNHD1</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">79</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">94</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SLFN13</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">81</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">86</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">NBPF26</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="summary">@summary</span> - Statistics about the samples</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">ID</th>
<th style="text-align: right;">summary</th>
<th style="text-align: right;">Mean</th>
<th style="text-align: right;">Median</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">NCBI_Build</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">Center</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Samples</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">nGenes</td>
<td style="text-align: right;">9576</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Frame_Shift_Del</td>
<td style="text-align: right;">2970</td>
<td style="text-align: right;">495.000</td>
<td style="text-align: right;">635.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Frame_Shift_Ins</td>
<td style="text-align: right;">2743</td>
<td style="text-align: right;">457.167</td>
<td style="text-align: right;">471.0</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="maf.silent">@maf.silent</span> - Table of all variants classified as silent mutations</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>maf.silent)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 51%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: left;">Chromosome</th>
<th style="text-align: right;">Start_Position</th>
<th style="text-align: right;">End_Position</th>
<th style="text-align: left;">Reference_Allele</th>
<th style="text-align: left;">Tumor_Seq_Allele2</th>
<th style="text-align: left;">Hugo_Symbol</th>
<th style="text-align: left;">Variant_Classification</th>
<th style="text-align: left;">tx</th>
<th style="text-align: left;">exon</th>
<th style="text-align: left;">txChange</th>
<th style="text-align: left;">aaChange</th>
<th style="text-align: left;">Variant_Type</th>
<th style="text-align: left;">Func.refGene</th>
<th style="text-align: left;">Gene.refGene</th>
<th style="text-align: left;">GeneDetail.refGene</th>
<th style="text-align: left;">ExonicFunc.refGene</th>
<th style="text-align: left;">AAChange.refGene</th>
<th style="text-align: left;">cytoBand</th>
<th style="text-align: left;">ExAC_ALL</th>
<th style="text-align: left;">ExAC_AFR</th>
<th style="text-align: left;">ExAC_AMR</th>
<th style="text-align: left;">ExAC_EAS</th>
<th style="text-align: left;">ExAC_FIN</th>
<th style="text-align: left;">ExAC_NFE</th>
<th style="text-align: left;">ExAC_OTH</th>
<th style="text-align: left;">ExAC_SAS</th>
<th style="text-align: left;">avsnp147</th>
<th style="text-align: left;">SIFT_score</th>
<th style="text-align: left;">SIFT_pred</th>
<th style="text-align: left;">Polyphen2_HDIV_score</th>
<th style="text-align: left;">Polyphen2_HDIV_pred</th>
<th style="text-align: left;">Polyphen2_HVAR_score</th>
<th style="text-align: left;">Polyphen2_HVAR_pred</th>
<th style="text-align: left;">LRT_score</th>
<th style="text-align: left;">LRT_pred</th>
<th style="text-align: left;">MutationTaster_score</th>
<th style="text-align: left;">MutationTaster_pred</th>
<th style="text-align: left;">MutationAssessor_score</th>
<th style="text-align: left;">MutationAssessor_pred</th>
<th style="text-align: left;">FATHMM_score</th>
<th style="text-align: left;">FATHMM_pred</th>
<th style="text-align: left;">PROVEAN_score</th>
<th style="text-align: left;">PROVEAN_pred</th>
<th style="text-align: left;">VEST3_score</th>
<th style="text-align: left;">CADD_raw</th>
<th style="text-align: left;">CADD_phred</th>
<th style="text-align: left;">DANN_score</th>
<th style="text-align: left;">fathmm-MKL_coding_score</th>
<th style="text-align: left;">fathmm-MKL_coding_pred</th>
<th style="text-align: left;">MetaSVM_score</th>
<th style="text-align: left;">MetaSVM_pred</th>
<th style="text-align: left;">MetaLR_score</th>
<th style="text-align: left;">MetaLR_pred</th>
<th style="text-align: left;">integrated_fitCons_score</th>
<th style="text-align: left;">integrated_confidence_value</th>
<th style="text-align: left;">GERP++_RS</th>
<th style="text-align: left;">phyloP7way_vertebrate</th>
<th style="text-align: left;">phyloP20way_mammalian</th>
<th style="text-align: left;">phastCons7way_vertebrate</th>
<th style="text-align: left;">phastCons20way_mammalian</th>
<th style="text-align: left;">SiPhy_29way_logOdds</th>
<th style="text-align: left;">Otherinfo1</th>
<th style="text-align: left;">Otherinfo2</th>
<th style="text-align: left;">Otherinfo3</th>
<th style="text-align: left;">Otherinfo4</th>
<th style="text-align: left;">Otherinfo5</th>
<th style="text-align: left;">Otherinfo6</th>
<th style="text-align: left;">Otherinfo7</th>
<th style="text-align: left;">Otherinfo8</th>
<th style="text-align: left;">Otherinfo9</th>
<th style="text-align: left;">Otherinfo10</th>
<th style="text-align: left;">Otherinfo11</th>
<th style="text-align: left;">Otherinfo12</th>
<th style="text-align: left;">Otherinfo13</th>
<th style="text-align: left;">ref_alt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">14455</td>
<td style="text-align: right;">14455</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">AAGAG</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">ncRNA_exonic</td>
<td style="text-align: left;">WASH7P</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">109.6</td>
<td style="text-align: left;">21</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">14455</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">TAAGAG</td>
<td style="text-align: left;">109.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-0.199;DP=22;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=5.22;ReadPosRankSum=2.335;SOR=0.074;ANN=TAAGAG|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>46_</em>47insAAGAG|||||47|,TAAGAG|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.<em>2913_</em>2914insCTCTT|||||2913|,TAAGAG|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|11/11|n.1299_1300insCTCTT||||||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:17,4:21:99:117,0,702</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">16665</td>
<td style="text-align: right;">16666</td>
<td style="text-align: left;">CG</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">ncRNA_exonic</td>
<td style="text-align: left;">WASH7P</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">37.6</td>
<td style="text-align: left;">15</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">16664</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">CCG</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">37.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.327;DP=15;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=2.51;ReadPosRankSum=-1.276;SOR=0.368;ANN=C|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|transcript|ENST00000456328.2|processed_transcript||n.<em>2256_</em>2257delCG|||||2256|,C|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|transcript|ENST00000619216.1|miRNA||n.<em>703_</em>704delCG|||||704|,C|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|transcript|ENST00000488147.1|unprocessed_pseudogene|8/11|n.1008_1009delCG||||||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:13,2:15:45:45,0,539</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">778406</td>
<td style="text-align: right;">778406</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">GCT</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">ncRNA_exonic</td>
<td style="text-align: left;">LOC100288069</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">rs142583823</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">756.6</td>
<td style="text-align: left;">45</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">778406</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">GGCT</td>
<td style="text-align: left;">756.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.389;DP=52;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=16.81;ReadPosRankSum=0.286;SOR=0.723;ANN=GGCT|upstream_gene_variant|MODIFIER|ENSG00000285268|ENSG00000285268|transcript|ENST00000644482.1|transcribed_processed_pseudogene||n.-3531_-3530insGCT|||||3530|,GGCT|intragenic_variant|MODIFIER|ENSG00000230021|ENSG00000230021|gene_variant|ENSG00000230021|||n.778409_778407dupAGC||||||,GGCT|non_coding_transcript_exon_variant|MODIFIER|ENSG00000228327|ENSG00000228327|transcript|ENST00000506640.2|processed_transcript|1/16|n.218_220dupAGC||||||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:25,20:45:99:764,0,990</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">842922</td>
<td style="text-align: right;">842922</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">CT</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">ncRNA_exonic</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">rs112119688</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">679.03</td>
<td style="text-align: left;">16</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">842922</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">CCT</td>
<td style="text-align: left;">679.03</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=2;AF=1.00;AN=2;DP=21;ExcessHet=0.0000;FS=0.000;MLEAC=2;MLEAF=1.00;MQ=60.00;QD=31.76;SOR=1.609;ANN=CCT|non_coding_transcript_exon_variant|MODIFIER|LINC01128|ENSG00000228794|transcript|ENST00000670780.1|pseudogene|3/8|n.2000_2001dupCT||||||INFO_REALIGN_3_PRIME</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">1/1:0,16:16:48:693,48,0</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">843252</td>
<td style="text-align: right;">843253</td>
<td style="text-align: left;">TT</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">ncRNA_exonic</td>
<td style="text-align: left;">LINC01128</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">40.6</td>
<td style="text-align: left;">14</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">843251</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">GTT</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">40.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-0.449;DP=15;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=2.90;ReadPosRankSum=0.183;SOR=0.693;ANN=G|non_coding_transcript_exon_variant|MODIFIER|LINC01128|ENSG00000228794|transcript|ENST00000670780.1|pseudogene|3/8|n.2329_2330delTT||||||INFO_REALIGN_3_PRIME</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:12,2:14:48:48,0,491</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: right;">914483</td>
<td style="text-align: right;">914483</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">ACTGCCCAGCTC</td>
<td style="text-align: left;">Unknown</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">intergenic</td>
<td style="text-align: left;">FAM41C;LINC02593</td>
<td style="text-align: left;">dist=37681;dist=2335</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">1p36.33</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">rs140295539</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">277.6</td>
<td style="text-align: left;">10</td>
<td style="text-align: left;">chr1</td>
<td style="text-align: left;">914483</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">GACTGCCCAGCTC</td>
<td style="text-align: left;">277.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-0.210;DP=13;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=27.76;ReadPosRankSum=2.638;SOR=0.527;ANN=GACTGCCCAGCTC|downstream_gene_variant|MODIFIER|LINC02593|ENSG00000223764|transcript|ENST00000609207.1|retained_intron||n.<em>2370_</em>2381dupGAGCTGGGCAGT|||||2381|,GACTGCCCAGCTC|intron_variant|MODIFIER|ENSG00000241180|ENSG00000241180|transcript|ENST00000398216.2|pseudogene|1/1|n.274+47_274+58dupAGCTCACTGCCC||||||INFO_REALIGN_3_PRIME,GACTGCCCAGCTC|intron_variant|MODIFIER|ENSG00000272438|ENSG00000272438|transcript|ENST00000607769.1|pseudogene|1/1|n.125-1259_125-1248dupAGCTCACTGCCC||||||INFO_REALIGN_3_PRIME,GACTGCCCAGCTC|non_coding_transcript_exon_variant|MODIFIER|ENSG00000230699|ENSG00000230699|transcript|ENST00000448179.1|pseudogene|2/2|n.2586_2597dupAGCTCACTGCCC||||||INFO_REALIGN_3_PRIME</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:3,7:10:99:285,0,105</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li><span class="citation" data-cites="clinical.data">@clinical.data</span> - Shows clinical data (like patient outcomes), if any. We don’t have any in this data.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ds.maf<span class="sc">@</span>clinical.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-CL</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-1-Pt</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-76-CL</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-76-Pt</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-81-CL</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-81-Pt</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="data-exploration" class="level2">
<h2 class="anchored" data-anchor-id="data-exploration">Data exploration</h2>
<p>These tables can be explored with typical tidyverse/dplyr functions to find things like specific mutations, shared mutations, unique mutations, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for mutations in a certain gene of interest</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>data <span class="sc">|&gt;</span> </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Hugo_Symbol <span class="sc">==</span> <span class="st">"JAK2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 17%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 24%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: left;">Chromosome</th>
<th style="text-align: right;">Start_Position</th>
<th style="text-align: right;">End_Position</th>
<th style="text-align: left;">Reference_Allele</th>
<th style="text-align: left;">Tumor_Seq_Allele2</th>
<th style="text-align: left;">Hugo_Symbol</th>
<th style="text-align: left;">Variant_Classification</th>
<th style="text-align: left;">tx</th>
<th style="text-align: left;">exon</th>
<th style="text-align: left;">txChange</th>
<th style="text-align: left;">aaChange</th>
<th style="text-align: left;">Variant_Type</th>
<th style="text-align: left;">Func.refGene</th>
<th style="text-align: left;">Gene.refGene</th>
<th style="text-align: left;">GeneDetail.refGene</th>
<th style="text-align: left;">ExonicFunc.refGene</th>
<th style="text-align: left;">AAChange.refGene</th>
<th style="text-align: left;">cytoBand</th>
<th style="text-align: left;">ExAC_ALL</th>
<th style="text-align: left;">ExAC_AFR</th>
<th style="text-align: left;">ExAC_AMR</th>
<th style="text-align: left;">ExAC_EAS</th>
<th style="text-align: left;">ExAC_FIN</th>
<th style="text-align: left;">ExAC_NFE</th>
<th style="text-align: left;">ExAC_OTH</th>
<th style="text-align: left;">ExAC_SAS</th>
<th style="text-align: left;">avsnp147</th>
<th style="text-align: left;">SIFT_score</th>
<th style="text-align: left;">SIFT_pred</th>
<th style="text-align: left;">Polyphen2_HDIV_score</th>
<th style="text-align: left;">Polyphen2_HDIV_pred</th>
<th style="text-align: left;">Polyphen2_HVAR_score</th>
<th style="text-align: left;">Polyphen2_HVAR_pred</th>
<th style="text-align: left;">LRT_score</th>
<th style="text-align: left;">LRT_pred</th>
<th style="text-align: left;">MutationTaster_score</th>
<th style="text-align: left;">MutationTaster_pred</th>
<th style="text-align: left;">MutationAssessor_score</th>
<th style="text-align: left;">MutationAssessor_pred</th>
<th style="text-align: left;">FATHMM_score</th>
<th style="text-align: left;">FATHMM_pred</th>
<th style="text-align: left;">PROVEAN_score</th>
<th style="text-align: left;">PROVEAN_pred</th>
<th style="text-align: left;">VEST3_score</th>
<th style="text-align: left;">CADD_raw</th>
<th style="text-align: left;">CADD_phred</th>
<th style="text-align: left;">DANN_score</th>
<th style="text-align: left;">fathmm-MKL_coding_score</th>
<th style="text-align: left;">fathmm-MKL_coding_pred</th>
<th style="text-align: left;">MetaSVM_score</th>
<th style="text-align: left;">MetaSVM_pred</th>
<th style="text-align: left;">MetaLR_score</th>
<th style="text-align: left;">MetaLR_pred</th>
<th style="text-align: left;">integrated_fitCons_score</th>
<th style="text-align: left;">integrated_confidence_value</th>
<th style="text-align: left;">GERP++_RS</th>
<th style="text-align: left;">phyloP7way_vertebrate</th>
<th style="text-align: left;">phyloP20way_mammalian</th>
<th style="text-align: left;">phastCons7way_vertebrate</th>
<th style="text-align: left;">phastCons20way_mammalian</th>
<th style="text-align: left;">SiPhy_29way_logOdds</th>
<th style="text-align: left;">Otherinfo1</th>
<th style="text-align: left;">Otherinfo2</th>
<th style="text-align: left;">Otherinfo3</th>
<th style="text-align: left;">Otherinfo4</th>
<th style="text-align: left;">Otherinfo5</th>
<th style="text-align: left;">Otherinfo6</th>
<th style="text-align: left;">Otherinfo7</th>
<th style="text-align: left;">Otherinfo8</th>
<th style="text-align: left;">Otherinfo9</th>
<th style="text-align: left;">Otherinfo10</th>
<th style="text-align: left;">Otherinfo11</th>
<th style="text-align: left;">Otherinfo12</th>
<th style="text-align: left;">Otherinfo13</th>
<th style="text-align: left;">ref_alt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: right;">5069059</td>
<td style="text-align: right;">5069059</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">Frame_Shift_Ins</td>
<td style="text-align: left;">NM_001322204</td>
<td style="text-align: left;">exon8</td>
<td style="text-align: left;">c.918dupA</td>
<td style="text-align: left;">p.N308Kfs*2</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">frameshift insertion</td>
<td style="text-align: left;">JAK2:NM_001322204:exon8:c.918dupA:p.N308Kfs<em>2,JAK2:NM_001322195:exon10:c.1365dupA:p.N457Kfs</em>2,JAK2:NM_001322196:exon10:c.1365dupA:p.N457Kfs<em>2,JAK2:NM_001322194:exon11:c.1365dupA:p.N457Kfs</em>2,JAK2:NM_001322198:exon11:c.150dupA:p.N52Kfs<em>2,JAK2:NM_001322199:exon11:c.150dupA:p.N52Kfs</em>2,JAK2:NM_004972:exon11:c.1365dupA:p.N457Kfs*2</td>
<td style="text-align: left;">9p24.1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">50.6</td>
<td style="text-align: left;">22</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: left;">5069059</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">CA</td>
<td style="text-align: left;">50.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.402;DP=23;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=2.30;ReadPosRankSum=0.300;SOR=0.368;ANN=CA|frameshift_variant|HIGH|JAK2|ENSG00000096968|transcript|ENST00000381652.4|protein_coding|11/25|c.1370dupA|p.Asn457fs|1838/7023|1371/3399|457/1132||INFO_REALIGN_3_PRIME;LOF=(JAK2|ENSG00000096968|1|1.00)</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:18,4:22:58:58,0,495</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-76-CL</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: right;">5078360</td>
<td style="text-align: right;">5078360</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_001322204</td>
<td style="text-align: left;">exon13</td>
<td style="text-align: left;">c.A1600G</td>
<td style="text-align: left;">p.R534G</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">JAK2:NM_001322204:exon13:c.A1600G:p.R534G,JAK2:NM_001322195:exon15:c.A2047G:p.R683G,JAK2:NM_001322196:exon15:c.A2047G:p.R683G,JAK2:NM_001322194:exon16:c.A2047G:p.R683G,JAK2:NM_001322198:exon16:c.A832G:p.R278G,JAK2:NM_001322199:exon16:c.A832G:p.R278G,JAK2:NM_004972:exon16:c.A2047G:p.R683G</td>
<td style="text-align: left;">9p24.1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.855</td>
<td style="text-align: left;">L</td>
<td style="text-align: left;">-1.67</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">-5.88</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.96</td>
<td style="text-align: left;">6.530</td>
<td style="text-align: left;">31</td>
<td style="text-align: left;">0.999</td>
<td style="text-align: left;">0.958</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.443</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.671</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.741</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">5.55</td>
<td style="text-align: left;">1.062</td>
<td style="text-align: left;">1.088</td>
<td style="text-align: left;">1.000</td>
<td style="text-align: left;">1.000</td>
<td style="text-align: left;">15.676</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">519.64</td>
<td style="text-align: left;">35</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: left;">5078360</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">519.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-3.865;DP=36;ExcessHet=0.0000;FS=1.476;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=14.85;ReadPosRankSum=0.037;SOR=1.136;ANN=G|missense_variant|MODERATE|JAK2|ENSG00000096968|transcript|ENST00000381652.4|protein_coding|16/25|c.2047A&gt;G|p.Arg683Gly|2514/7023|2047/3399|683/1132||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:10,25:35:99:527,0,210</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DS-81-CL</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: right;">5065032</td>
<td style="text-align: right;">5065032</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">Frame_Shift_Del</td>
<td style="text-align: left;">NM_001322204</td>
<td style="text-align: left;">exon6</td>
<td style="text-align: left;">c.759delC</td>
<td style="text-align: left;">p.P254Qfs*11</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">frameshift deletion</td>
<td style="text-align: left;">JAK2:NM_001322204:exon6:c.759delC:p.P254Qfs<em>11,JAK2:NM_001322195:exon8:c.1206delC:p.P403Qfs</em>11,JAK2:NM_001322196:exon8:c.1206delC:p.P403Qfs<em>11,JAK2:NM_001322194:exon9:c.1206delC:p.P403Qfs</em>11,JAK2:NM_004972:exon9:c.1206delC:p.P403Qfs*11</td>
<td style="text-align: left;">9p24.1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">55.6</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: left;">5065031</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">GC</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">55.6</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-0.812;DP=9;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=6.18;ReadPosRankSum=0.812;SOR=0.495;ANN=G|frameshift_variant|HIGH|JAK2|ENSG00000096968|transcript|ENST00000381652.4|protein_coding|9/25|c.1208delC|p.Pro403fs|1675/7023|1208/3399|403/1132||INFO_REALIGN_3_PRIME;LOF=(JAK2|ENSG00000096968|1|1.00)</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:7,2:9:63:63,0,288</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">DS-81-CL</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: right;">5065033</td>
<td style="text-align: right;">5065033</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_001322204</td>
<td style="text-align: left;">exon6</td>
<td style="text-align: left;">c.C760T</td>
<td style="text-align: left;">p.P254S</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">JAK2</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">JAK2:NM_001322204:exon6:c.C760T:p.P254S,JAK2:NM_001322195:exon8:c.C1207T:p.P403S,JAK2:NM_001322196:exon8:c.C1207T:p.P403S,JAK2:NM_001322194:exon9:c.C1207T:p.P403S,JAK2:NM_004972:exon9:c.C1207T:p.P403S</td>
<td style="text-align: left;">9p24.1</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1.0</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">2.725</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">-0.04</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">-7.17</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.809</td>
<td style="text-align: left;">6.505</td>
<td style="text-align: left;">31</td>
<td style="text-align: left;">0.999</td>
<td style="text-align: left;">0.981</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.108</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">0.496</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">0.732</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">5.15</td>
<td style="text-align: left;">0.871</td>
<td style="text-align: left;">0.876</td>
<td style="text-align: left;">1.000</td>
<td style="text-align: left;">1.000</td>
<td style="text-align: left;">18.619</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">55.64</td>
<td style="text-align: left;">9</td>
<td style="text-align: left;">chr9</td>
<td style="text-align: left;">5065033</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">55.64</td>
<td style="text-align: left;">PASS</td>
<td style="text-align: left;">AC=1;AF=0.500;AN=2;BaseQRankSum=-1.383;DP=9;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=6.18;ReadPosRankSum=0.967;SOR=0.495;ANN=T|missense_variant|MODERATE|JAK2|ENSG00000096968|transcript|ENST00000381652.4|protein_coding|9/25|c.1207C&gt;T|p.Pro403Ser|1674/7023|1207/3399|403/1132||</td>
<td style="text-align: left;">GT:AD:DP:GQ:PL</td>
<td style="text-align: left;">0/1:7,2:9:63:63,0,288</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Dallas gave me a list of commonly mutated genes in BALL samples. I can load that in and search the whole list at once using <code>inner_join()</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for variants in a list of genes</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in data</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>dallas.df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">"data/common_mutated_genes.csv"</span>,</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">header =</span> F,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col.names =</span> <span class="st">"gene_symbol"</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dallas.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">gene_symbol</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">TYK2</td>
</tr>
<tr class="even">
<td style="text-align: left;">FLT3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">IL7R</td>
</tr>
<tr class="even">
<td style="text-align: left;">IKZF1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TCF3</td>
</tr>
<tr class="even">
<td style="text-align: left;">EBF1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>dallas.df <span class="sc">|&gt;</span> </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(ds.maf<span class="sc">@</span>data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_symbol"</span> <span class="ot">=</span> <span class="st">"Hugo_Symbol"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene_symbol<span class="sc">:</span>AAChange.refGene) <span class="sc">|&gt;</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 6%">
<col style="width: 7%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 7%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">gene_symbol</th>
<th style="text-align: left;">Tumor_Sample_Barcode</th>
<th style="text-align: left;">Chromosome</th>
<th style="text-align: right;">Start_Position</th>
<th style="text-align: right;">End_Position</th>
<th style="text-align: left;">Reference_Allele</th>
<th style="text-align: left;">Tumor_Seq_Allele2</th>
<th style="text-align: left;">Variant_Classification</th>
<th style="text-align: left;">tx</th>
<th style="text-align: left;">exon</th>
<th style="text-align: left;">txChange</th>
<th style="text-align: left;">aaChange</th>
<th style="text-align: left;">Variant_Type</th>
<th style="text-align: left;">Func.refGene</th>
<th style="text-align: left;">Gene.refGene</th>
<th style="text-align: left;">GeneDetail.refGene</th>
<th style="text-align: left;">ExonicFunc.refGene</th>
<th style="text-align: left;">AAChange.refGene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28018498</td>
<td style="text-align: right;">28018500</td>
<td style="text-align: left;">ATG</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">In_Frame_Del</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon20</td>
<td style="text-align: left;">c.2508_2510del</td>
<td style="text-align: left;">p.I836del</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonframeshift deletion</td>
<td style="text-align: left;">FLT3:NM_004119:exon20:c.2508_2510del:p.I836del</td>
</tr>
<tr class="even">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28050157</td>
<td style="text-align: right;">28050157</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon6</td>
<td style="text-align: left;">c.C680T</td>
<td style="text-align: left;">p.T227M</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">FLT3:NM_004119:exon6:c.C680T:p.T227M</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-CL</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28100491</td>
<td style="text-align: right;">28100491</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon1</td>
<td style="text-align: left;">c.A20G</td>
<td style="text-align: left;">p.D7G</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">FLT3:NM_004119:exon1:c.A20G:p.D7G</td>
</tr>
<tr class="even">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28015161</td>
<td style="text-align: right;">28015161</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon22</td>
<td style="text-align: left;">c.G2749A</td>
<td style="text-align: left;">p.E917K</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">FLT3:NM_004119:exon22:c.G2749A:p.E917K</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28018498</td>
<td style="text-align: right;">28018500</td>
<td style="text-align: left;">ATG</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">In_Frame_Del</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon20</td>
<td style="text-align: left;">c.2508_2510del</td>
<td style="text-align: left;">p.I836del</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonframeshift deletion</td>
<td style="text-align: left;">FLT3:NM_004119:exon20:c.2508_2510del:p.I836del</td>
</tr>
<tr class="even">
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">DS-1-Pt</td>
<td style="text-align: left;">chr13</td>
<td style="text-align: right;">28050157</td>
<td style="text-align: right;">28050157</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">Missense_Mutation</td>
<td style="text-align: left;">NM_004119</td>
<td style="text-align: left;">exon6</td>
<td style="text-align: left;">c.C680T</td>
<td style="text-align: left;">p.T227M</td>
<td style="text-align: left;">SNP</td>
<td style="text-align: left;">exonic</td>
<td style="text-align: left;">FLT3</td>
<td style="text-align: left;">.</td>
<td style="text-align: left;">nonsynonymous SNV</td>
<td style="text-align: left;">FLT3:NM_004119:exon6:c.C680T:p.T227M</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>One more example: What gene has the most mutations? Do samples share the same mutations in a gene?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for gene with the most mutations</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>gene.summary <span class="sc">|&gt;</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(total)) <span class="sc">|&gt;</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hugo_Symbol, total) <span class="sc">|&gt;</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Hugo_Symbol</th>
<th style="text-align: right;">total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ZC3H11B</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="even">
<td style="text-align: left;">TAF1L</td>
<td style="text-align: right;">165</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OBSCN</td>
<td style="text-align: right;">151</td>
</tr>
<tr class="even">
<td style="text-align: left;">RGPD1</td>
<td style="text-align: right;">118</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TTN</td>
<td style="text-align: right;">116</td>
</tr>
<tr class="even">
<td style="text-align: left;">RGPD3</td>
<td style="text-align: right;">116</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at the gene amino acid changes that the samples share</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>data <span class="sc">|&gt;</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Hugo_Symbol <span class="sc">==</span> <span class="st">"ZC3H11B"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(aaChange) <span class="sc">|&gt;</span> </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span> </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(count)) <span class="sc">|&gt;</span> </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">aaChange</th>
<th style="text-align: right;">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">p.A628G</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">p.E320K</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p.E413K</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">p.E414K</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p.G279S</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">p.H211N</td>
<td style="text-align: right;">4</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Four of the six samples share some variants. That makes some sense, since the samples are 3 bone marrow biopsy samples of Down Syndrome patients with BALL and the cell lines derived from those samples</p>
</section>
<section id="side-tangent-different-transcripts-can-give-multiple-variant-results" class="level2">
<h2 class="anchored" data-anchor-id="side-tangent-different-transcripts-can-give-multiple-variant-results">SIDE TANGENT: Different transcripts can give multiple variant results!</h2>
<p>I noticed an issue with some of my results. From Dallas, and from loading the bam/vcf files into IGV software, I know that the DS-76-CL sample has a JAK2 R683G mutation. This figure comes from loading the .bam file of DS-76-CL into IGV:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/igv%20view.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>This is the IGV info on the mutation:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/igv%20annotation.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>However, when looking at the <code>aaChange</code> variable in my data, I get something else: R534G.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>data <span class="sc">|&gt;</span> </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Hugo_Symbol <span class="sc">==</span> <span class="st">"JAK2"</span>,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>         Tumor_Sample_Barcode <span class="sc">==</span> <span class="st">"DS-76-CL"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(aaChange)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">aaChange</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">p.R534G</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>After some digging, it turns out I have multiple transcripts for this variant, stored in the <code>AAChange.refGene</code> column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>data <span class="sc">|&gt;</span> </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Hugo_Symbol <span class="sc">==</span> <span class="st">"JAK2"</span>,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>         Tumor_Sample_Barcode <span class="sc">==</span> <span class="st">"DS-76-CL"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(AAChange.refGene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">AAChange.refGene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">JAK2:NM_001322204:exon13:c.A1600G:p.R534G,JAK2:NM_001322195:exon15:c.A2047G:p.R683G,JAK2:NM_001322196:exon15:c.A2047G:p.R683G,JAK2:NM_001322194:exon16:c.A2047G:p.R683G,JAK2:NM_001322198:exon16:c.A832G:p.R278G,JAK2:NM_001322199:exon16:c.A832G:p.R278G,JAK2:NM_004972:exon16:c.A2047G:p.R683G</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Seven transcript variants, actually:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>ds.maf<span class="sc">@</span>data <span class="sc">|&gt;</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Hugo_Symbol <span class="sc">==</span> <span class="st">"JAK2"</span>,</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>         Tumor_Sample_Barcode <span class="sc">==</span> <span class="st">"DS-76-CL"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(AAChange.refGene) <span class="sc">|&gt;</span> </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_longer_delim</span>(<span class="at">cols =</span> AAChange.refGene, <span class="at">delim =</span> <span class="st">","</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">AAChange.refGene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">JAK2:NM_001322204:exon13:c.A1600G:p.R534G</td>
</tr>
<tr class="even">
<td style="text-align: left;">JAK2:NM_001322195:exon15:c.A2047G:p.R683G</td>
</tr>
<tr class="odd">
<td style="text-align: left;">JAK2:NM_001322196:exon15:c.A2047G:p.R683G</td>
</tr>
<tr class="even">
<td style="text-align: left;">JAK2:NM_001322194:exon16:c.A2047G:p.R683G</td>
</tr>
<tr class="odd">
<td style="text-align: left;">JAK2:NM_001322198:exon16:c.A832G:p.R278G</td>
</tr>
<tr class="even">
<td style="text-align: left;">JAK2:NM_001322199:exon16:c.A832G:p.R278G</td>
</tr>
<tr class="odd">
<td style="text-align: left;">JAK2:NM_004972:exon16:c.A2047G:p.R683G</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>How does ANNOVAR handle alternative transcripts? I looked into the FAQ:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/splice%20variant%20FAQ.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>When there are multiple transcripts for a variant, ANNOVAR has an order of precedence that it uses to decide which one to put in <code>aaChange</code> and which ones to store in <code>AAChange.refGene</code>. Here is the order:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="2024_05_03_figures/transcript%20precedence.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>Unfortunately, all of ours are exonic and have “equal” precedence. I think the next order of precedence is the lowest exon, which results in our <code>aaChange</code> being R534G, since exon13 comes before exon15 and exon16.</p>
<p>The way to fix this? I am still troubleshooting it, but there is an argument called <code>--separate</code> which will put each transcript variant result in its own row. Unfortunately, this can’t be called directly with <code>table_annovar.pl.</code> I would need to use <code>annotate_variation.pl</code>, which is a more in-depth script that can lead to mistakes if you don’t know what you’re doing.</p>
<p>There may be a way to specify the <code>--separate</code> argument within the <code>table_annovar.pl</code> command, but I have not had the time to tinker yet. Check back with this Notion page in the future and I should have a solution posted.</p>
<p>For now, we can move on to the more gratifying part: using maftools for visualization.</p>
</section>
<section id="maftools-plots-and-analysis" class="level2">
<h2 class="anchored" data-anchor-id="maftools-plots-and-analysis">maftools : plots and analysis</h2>
<p>Many of the plots below were generated following the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html#1_Introduction">maftools vignette</a>. I have commented out the parts of the code that specify plot features and saves the plot for display purposes. I left them in because, in my current hands, I could not get the plots to work with any sort of <code>ggsave()</code> function and it was the best way to save the plots I knew of.</p>
</section>
<section id="summary-plot" class="level2">
<h2 class="anchored" data-anchor-id="summary-plot">Summary plot</h2>
<p>This is the summary plot, which has a few things combined. Counts for variant classification, type, Single Nucleotide Variant (SNV) class, variants per sample, and variant classification per sample. Also a list of the top ten mutated genes. Stacked bar charts denote the types of mutations in the gene, with the Variant Classification graph providing the color legend for each variant type. The percentage at the end of the bar indicates the number of samples in the MAF object that also contain a variant in that gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># png(filename = paste0(results.dir, "variant_summary_plot.png"),</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     width = 11,</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     height = 8,</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     units = "in",</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     res = 300,</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     bg = "white")</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotmafSummary</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">showBarcodes =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in titv(maf = maf, useSyn = TRUE, plot = FALSE): Non standard Ti/Tv
class: 368TRUE</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>NOTE: The top 10 mutated plot and the oncoplot below seem to prioritize sample number over number of mutations when defining “top mutated genes.” You’ll notice that ZC3H11B and TAF1L gene from our earlier code chunk looking at “What gene has the most mutations?” is not on the plot.</p>
</section>
<section id="oncoplot" class="level2">
<h2 class="anchored" data-anchor-id="oncoplot">Oncoplot</h2>
<p>The <code>oncoplot()</code> function shows a larger view of a group of genes and their variant types. There is a <a href="https://bioconductor.org/packages/devel/bioc/vignettes/maftools/inst/doc/oncoplots.html">separate vignette on customizing these plots</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoplot</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">top =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in min(x): no non-missing arguments to min; returning Inf</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in max(x): no non-missing arguments to max; returning -Inf</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Most of these genes have more than one mutation (usually 2 or more missense mutations), so they show up as black.</p>
</section>
<section id="transitions-and-transversions" class="level2">
<h2 class="anchored" data-anchor-id="transitions-and-transversions">Transitions and Transversions</h2>
<p>This plot shows the breakdown of the single nucleotide polymorphisms (SNPs) and whether they are transitions or transversions.</p>
<p>Transitions are changes within purine or pyrimidine nucleotide type. A &lt;-&gt; G or C &lt;-&gt; T. Transversions are changes of purine &lt;-&gt; pyrimidine</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>maf.titv <span class="ot">&lt;-</span> <span class="fu">titv</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="lollipop-plots" class="level2">
<h2 class="anchored" data-anchor-id="lollipop-plots">Lollipop plots</h2>
<p>A “Lollipop Plot” lets you look at mutations on a gene of interest. The higher the lollipop, the more samples share the mutation.</p>
<p>I’ll output two versions for my JAK2 variant, one using <code>aaChange</code> and the other using <code>AAChange.refGene</code> to show the difference the transcript variants can make</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># png(filename = paste0(results.dir, "JAK2_lollipop.png"),</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     width = 6,</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     height = 4,</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     units = "in",</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     res = 300,</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     bg = "white")</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot using aaChange</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="fu">lollipopPlot</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">gene =</span> <span class="st">"JAK2"</span>,</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">AACol =</span> <span class="st">"aaChange"</span>,</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">showMutationRate =</span> T,</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">labelPos =</span> <span class="st">"all"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Plot using AAChange.refGene</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot using aaChange</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lollipopPlot</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">gene =</span> <span class="st">"JAK2"</span>,</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">AACol =</span> <span class="st">"AAChange.refGene"</span>,</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">showMutationRate =</span> T,</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">labelPos =</span> <span class="st">"all"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>NOTE: To generate a plot, your MAF object must have amino acid change information. There may be several naming conventions for this column. By default, the function looks for a column called <code>AAChange</code>. I can specify <code>aaChange</code> or <code>AAChange.refGene</code> for my data.</p>
<p>The <code>getFields()</code> function shows the available fields, but take a look at your MAF object to confirm you have them and what data they hold.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getFields</span>(ds.maf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Tumor_Sample_Barcode"        "Chromosome"                 
 [3] "Start_Position"              "End_Position"               
 [5] "Reference_Allele"            "Tumor_Seq_Allele2"          
 [7] "Hugo_Symbol"                 "Variant_Classification"     
 [9] "tx"                          "exon"                       
[11] "txChange"                    "aaChange"                   
[13] "Variant_Type"                "Func.refGene"               
[15] "Gene.refGene"                "GeneDetail.refGene"         
[17] "ExonicFunc.refGene"          "AAChange.refGene"           
[19] "cytoBand"                    "ExAC_ALL"                   
[21] "ExAC_AFR"                    "ExAC_AMR"                   
[23] "ExAC_EAS"                    "ExAC_FIN"                   
[25] "ExAC_NFE"                    "ExAC_OTH"                   
[27] "ExAC_SAS"                    "avsnp147"                   
[29] "SIFT_score"                  "SIFT_pred"                  
[31] "Polyphen2_HDIV_score"        "Polyphen2_HDIV_pred"        
[33] "Polyphen2_HVAR_score"        "Polyphen2_HVAR_pred"        
[35] "LRT_score"                   "LRT_pred"                   
[37] "MutationTaster_score"        "MutationTaster_pred"        
[39] "MutationAssessor_score"      "MutationAssessor_pred"      
[41] "FATHMM_score"                "FATHMM_pred"                
[43] "PROVEAN_score"               "PROVEAN_pred"               
[45] "VEST3_score"                 "CADD_raw"                   
[47] "CADD_phred"                  "DANN_score"                 
[49] "fathmm-MKL_coding_score"     "fathmm-MKL_coding_pred"     
[51] "MetaSVM_score"               "MetaSVM_pred"               
[53] "MetaLR_score"                "MetaLR_pred"                
[55] "integrated_fitCons_score"    "integrated_confidence_value"
[57] "GERP++_RS"                   "phyloP7way_vertebrate"      
[59] "phyloP20way_mammalian"       "phastCons7way_vertebrate"   
[61] "phastCons20way_mammalian"    "SiPhy_29way_logOdds"        
[63] "Otherinfo1"                  "Otherinfo2"                 
[65] "Otherinfo3"                  "Otherinfo4"                 
[67] "Otherinfo5"                  "Otherinfo6"                 
[69] "Otherinfo7"                  "Otherinfo8"                 
[71] "Otherinfo9"                  "Otherinfo10"                
[73] "Otherinfo11"                 "Otherinfo12"                
[75] "Otherinfo13"                 "ref_alt"                    </code></pre>
</div>
</div>
</section>
<section id="mafbarplot" class="level2">
<h2 class="anchored" data-anchor-id="mafbarplot">mafbarplot()</h2>
<p>Similar to the oncoplot, but the bar plot takes a count of total mutations and plots it on the x-axis. It’s also a stacked bar plot, where the bar for each gene is seperated into chunks denoting mutation type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># png(filename = paste0(results.dir, "variant_bar_plot.png"),</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     width = 11,</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     height = 8,</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     units = "in",</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     res = 300,</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     bg = "white")</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mafbarplot</span>(ds.maf, <span class="at">n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rainfall-plots" class="level2">
<h2 class="anchored" data-anchor-id="rainfall-plots">Rainfall plots</h2>
<p>This plot displays variants as dots on a y-axis that plots inter variant distance. Cancers are known for localized hyper-mutations, so this plot would be good for identifying those locations. My data apparently has a lot of them, which I’m not sure how to interpret, but I wanted to demonstrate the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">subsetMaf</span>(<span class="at">maf =</span> ds.maf, <span class="at">tsb =</span> <span class="st">"DS-81-Pt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>--Possible FLAGS among top ten genes:
  OBSCN
-Processing clinical data</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rainfallPlot</span>(<span class="at">maf =</span> temp, <span class="at">detectChangePoints =</span> T, <span class="at">pointSize =</span> <span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing DS-81-Pt..</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Kataegis detected at:</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>      Chromosome Start_Position End_Position nMuts Avg_intermutation_dist Size
   1:          1         631811       634798    11              298.70000 2987
   2:          1         851992       857100    11              510.80000 5108
   3:          1         913065       920728    21              383.15000 7663
   4:          1         952421       958790     9              796.12500 6369
   5:          1        1054900      1055604     7              117.33333  704
  ---                                                                         
1372:         23      123909214    123913445    19              235.05556 4231
1373:         23      150231160    150231845    14               52.69231  685
1374:         23      154469313    154476336    13              585.25000 7023
1375:         23      156020871    156025116    16              283.00000 4245
1376:         24       13286769     13287351    11               58.20000  582
      Tumor_Sample_Barcode C&gt;A C&gt;T T&gt;C C&gt;G T&gt;A T&gt;G NA
   1:             DS-81-Pt   2   5   4  NA  NA  NA NA
   2:             DS-81-Pt  NA   7   2   2  NA  NA NA
   3:             DS-81-Pt   1   8   6   3   1   2 NA
   4:             DS-81-Pt   1   3   4   1  NA  NA NA
   5:             DS-81-Pt  NA   6   1  NA  NA  NA NA
  ---                                                
1372:             DS-81-Pt  NA  NA  19  NA  NA  NA NA
1373:             DS-81-Pt  NA   7   7  NA  NA  NA NA
1374:             DS-81-Pt  NA   2   9   1   1  NA NA
1375:             DS-81-Pt   1   7   6   1   1  NA NA
1376:             DS-81-Pt   3   1   4  NA   1   2 NA</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="comparing-mutation-load-against-tcga-cohorts" class="level2">
<h2 class="anchored" data-anchor-id="comparing-mutation-load-against-tcga-cohorts">Comparing mutation load against TCGA cohorts</h2>
<p>Uses mutation load data from TCGA to compare MAF object against.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>ds.maf.mutload <span class="ot">&lt;-</span> <span class="fu">tcgaCompare</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">cohortName =</span> <span class="st">"DS MAF"</span>,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">logscale =</span> T,</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">capture_size =</span> <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in FUN(X[[i]], ...): Removed 0 samples with zero mutations.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Capture size [TCGA]:  35.8</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Capture size [Input]: 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Performing pairwise t-test for differences in mutation burden (per MB)..</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="copy-number-data-and-more" class="level2">
<h2 class="anchored" data-anchor-id="copy-number-data-and-more">Copy number data and more</h2>
<p>I don’t believe I used a library to analyze copy number data, so I don’t have the right fields in my MAF object to generate the copy number data plots. I encourage looking at the vignette to see additional maftools visualization and functionality.</p>
</section>
<section id="other-plots" class="level2">
<h2 class="anchored" data-anchor-id="other-plots">Other plots</h2>
<p>Drug Interaction plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">drugInteractions</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">fontSize =</span> <span class="fl">0.75</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 6%">
<col style="width: 44%">
<col style="width: 28%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: left;">gene_long_name</th>
<th style="text-align: left;">category_sources</th>
<th style="text-align: left;">category</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CENPF</td>
<td style="text-align: left;">CENTROMERE PROTEIN F</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TUMOR SUPPRESSOR</td>
</tr>
<tr class="even">
<td style="text-align: left;">CENPF</td>
<td style="text-align: left;">CENTROMERE PROTEIN F</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">DRUG RESISTANCE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CENPF</td>
<td style="text-align: left;">CENTROMERE PROTEIN F</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TRANSCRIPTION FACTOR BINDING</td>
</tr>
<tr class="even">
<td style="text-align: left;">PARP4</td>
<td style="text-align: left;">POLY(ADP-RIBOSE) POLYMERASE FAMILY MEMBER 4</td>
<td style="text-align: left;">HopkinsGroom,RussLampel,HingoraniCasas</td>
<td style="text-align: left;">DRUGGABLE GENOME</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PARP4</td>
<td style="text-align: left;">POLY(ADP-RIBOSE) POLYMERASE FAMILY MEMBER 4</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">RNA DIRECTED DNA POLYMERASE</td>
</tr>
<tr class="even">
<td style="text-align: left;">PARP4</td>
<td style="text-align: left;">POLY(ADP-RIBOSE) POLYMERASE FAMILY MEMBER 4</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">DNA REPAIR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PARP4</td>
<td style="text-align: left;">POLY(ADP-RIBOSE) POLYMERASE FAMILY MEMBER 4</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">DRUG RESISTANCE</td>
</tr>
<tr class="even">
<td style="text-align: left;">HLA-DPA1</td>
<td style="text-align: left;">MAJOR HISTOCOMPATIBILITY COMPLEX, CLASS II, DP ALPHA 1</td>
<td style="text-align: left;">HingoraniCasas</td>
<td style="text-align: left;">DRUGGABLE GENOME</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HLA-DPA1</td>
<td style="text-align: left;">MAJOR HISTOCOMPATIBILITY COMPLEX, CLASS II, DP ALPHA 1</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">CELL SURFACE</td>
</tr>
<tr class="even">
<td style="text-align: left;">MKI67</td>
<td style="text-align: left;">MARKER OF PROLIFERATION KI-67</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TUMOR SUPPRESSOR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LAMA5</td>
<td style="text-align: left;">LAMININ SUBUNIT ALPHA 5</td>
<td style="text-align: left;">HingoraniCasas</td>
<td style="text-align: left;">DRUGGABLE GENOME</td>
</tr>
<tr class="even">
<td style="text-align: left;">TTN</td>
<td style="text-align: left;">TITIN</td>
<td style="text-align: left;">HingoraniCasas,dGene,HopkinsGroom</td>
<td style="text-align: left;">DRUGGABLE GENOME</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TTN</td>
<td style="text-align: left;">TITIN</td>
<td style="text-align: left;">dGene,GO,GuideToPharmacologyGenes</td>
<td style="text-align: left;">SERINE THREONINE KINASE</td>
</tr>
<tr class="even">
<td style="text-align: left;">TTN</td>
<td style="text-align: left;">TITIN</td>
<td style="text-align: left;">GO,GuideToPharmacologyGenes,HopkinsGroom</td>
<td style="text-align: left;">KINASE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TTN</td>
<td style="text-align: left;">TITIN</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TYROSINE KINASE</td>
</tr>
<tr class="even">
<td style="text-align: left;">ALMS1</td>
<td style="text-align: left;">ALMS1, CENTROSOME AND BASAL BODY ASSOCIATED PROTEIN</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TUMOR SUPPRESSOR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OBSCN</td>
<td style="text-align: left;">OBSCURIN, CYTOSKELETAL CALMODULIN AND TITIN-INTERACTING RHOGEF</td>
<td style="text-align: left;">HopkinsGroom,GO,GuideToPharmacologyGenes</td>
<td style="text-align: left;">KINASE</td>
</tr>
<tr class="even">
<td style="text-align: left;">OBSCN</td>
<td style="text-align: left;">OBSCURIN, CYTOSKELETAL CALMODULIN AND TITIN-INTERACTING RHOGEF</td>
<td style="text-align: left;">HopkinsGroom,HingoraniCasas,dGene</td>
<td style="text-align: left;">DRUGGABLE GENOME</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OBSCN</td>
<td style="text-align: left;">OBSCURIN, CYTOSKELETAL CALMODULIN AND TITIN-INTERACTING RHOGEF</td>
<td style="text-align: left;">GO,GuideToPharmacologyGenes,dGene</td>
<td style="text-align: left;">SERINE THREONINE KINASE</td>
</tr>
<tr class="even">
<td style="text-align: left;">AHNAK</td>
<td style="text-align: left;">AHNAK NUCLEOPROTEIN</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">ION CHANNEL</td>
</tr>
<tr class="odd">
<td style="text-align: left;">AHNAK</td>
<td style="text-align: left;">AHNAK NUCLEOPROTEIN</td>
<td style="text-align: left;">GO</td>
<td style="text-align: left;">TRANSPORTER</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Breakdown of pathways by variant count</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pathways</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">plotType =</span> <span class="st">"treemap"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Summarizing signalling pathways [Sanchez-Vega et al., https://doi.org/10.1016/j.cell.2018.03.035]</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 3%">
<col style="width: 18%">
<col style="width: 20%">
<col style="width: 17%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Pathway</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">n_affected_genes</th>
<th style="text-align: right;">fraction_affected</th>
<th style="text-align: right;">Mutated_samples</th>
<th style="text-align: right;">Fraction_mutated_samples</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">TP53</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.5000000</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">MYC</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0.6923077</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PI3K</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">0.3793103</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">Hippo</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.5526316</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WNT</td>
<td style="text-align: right;">68</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">0.5294118</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">NOTCH</td>
<td style="text-align: right;">71</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">0.5915493</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RTK-RAS</td>
<td style="text-align: right;">85</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">0.5764706</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">NRF2</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.6666667</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.8333333</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TGF-Beta</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.4285714</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.8333333</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cell_Cycle</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.3333333</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.8333333</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>pws <span class="ot">&lt;-</span> <span class="fu">pathways</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">plotType =</span> <span class="st">"bar"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Summarizing signalling pathways [Sanchez-Vega et al., https://doi.org/10.1016/j.cell.2018.03.035]</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-37-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>More Pathway plots</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPathways</span>(<span class="at">maf =</span> ds.maf,</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">pathlist =</span> pws)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_05_03_VCF_Code_Review_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>That pretty much covers the plots I was able to make with my MAF object. But if you have clinical data, copy number data, or other you can generate even more plots using the vignette.</p>
</section>
</section>
<section id="sessioninfo" class="level1">
<h1>sessionInfo()</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Denver
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] vcfR_1.14.0     maftools_2.18.0 lubridate_1.9.2 forcats_1.0.0  
 [5] stringr_1.5.0   dplyr_1.1.2     purrr_1.0.2     readr_2.1.4    
 [9] tidyr_1.3.0     tibble_3.2.1    ggplot2_3.4.4   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] gtable_0.3.4       xfun_0.39          htmlwidgets_1.6.2  lattice_0.21-8    
 [5] tzdb_0.4.0         vctrs_0.6.3        tools_4.3.1        generics_0.1.3    
 [9] parallel_4.3.1     fansi_1.0.4        cluster_2.1.4      pkgconfig_2.0.3   
[13] R.oo_1.25.0        Matrix_1.6-1       data.table_1.14.8  RColorBrewer_1.1-3
[17] lifecycle_1.0.3    compiler_4.3.1     munsell_0.5.0      codetools_0.2-19  
[21] permute_0.9-7      htmltools_0.5.5    yaml_2.3.7         pillar_1.9.0      
[25] MASS_7.3-60        R.utils_2.12.3     vegan_2.6-4        nlme_3.1-162      
[29] tidyselect_1.2.0   digest_0.6.33      stringi_1.7.12     splines_4.3.1     
[33] fastmap_1.1.1      grid_4.3.1         colorspace_2.1-0   cli_3.6.1         
[37] magrittr_2.0.3     survival_3.5-5     utf8_1.2.3         ape_5.7-1         
[41] withr_2.5.0        scales_1.2.1       timechange_0.2.0   rmarkdown_2.25    
[45] R.methodsS3_1.8.2  hms_1.1.3          memuse_4.2-3       DNAcopy_1.74.1    
[49] evaluate_0.21      knitr_1.43         viridisLite_0.4.2  mgcv_1.8-42       
[53] rlang_1.1.1        Rcpp_1.0.11        pinfsc50_1.2.0     glue_1.6.2        
[57] rstudioapi_0.15.0  jsonlite_1.8.7     R6_2.5.1          </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>